sycl/test/basic_tests/handler/handler_mem_op.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clang -std=c++11 -fsycl %s -o %t.out -lstdc++ -lOpenCL -lsycl
+// RUN: env ENABLE_INFER_AS=1 %clang -std=c++11 -fsycl %s -o %t.out -lstdc++ -lOpenCL -lsycl
```

Comment:
AFAIK, this variable is not needed.

Reply:
It is.  Adding pointers requires this mode to be enabled in the compiler.  Otherwise conversations between global/generic aren't properly handled.

Reply:
@asavonic, could you confirm that ENABLE_INFER_AS is required here, please?
I expected https://github.com/intel/llvm/pull/242 to fix this case.